prov/efa/src/rxr/rxr_ep.c

```diff
@@ -1120,44 +1120,28 @@ static ssize_t rxr_ep_cancel(fid_t fid_ep, void *context)
  */
 static int efa_set_fi_hmem_p2p_opt(struct efa_ep *efa_ep, int opt)
 {
-	struct efa_hmem_info *hmem_info;
+	int i, ret;
 
-	hmem_info = efa_ep->domain->hmem_info;
-
-	switch (opt) {
+	/* FI_HMEM interfaces to check against */
+	static enum fi_hmem_iface ifaces[] = {
+		FI_HMEM_CUDA,
+		FI_HMEM_NEURON,
+		FI_HMEM_SYNAPSEAI
+	};
```

Comment:
We did not validate against `FI_HMEM_SYNAPSEAI` previously. I assume we should do so.

Also not sure about `FI_HMEM_SYSTEM`.